### PR TITLE
js/go sdks: use structs for arguments & remove tryConnect function

### PR DESCRIPTION
**Pull request by harjotgill**
## Summary by CodeRabbit

- **New Features**
  - Introduced `MiddlewareParams` and `FlowParams` structures to streamline the configuration of middleware and flow initiation.
  - Added an `Options` struct for more flexible client configuration.

- **Improvements**
  - Consolidated multiple individual parameters into structured parameters for better maintainability and readability.
  - Updated middleware setup to utilize new parameter structures, enhancing the setup process.

- **Refactor**
  - Refactored function signatures to accept new structured parameters.
  - Simplified error handling by removing unnecessary imports and code blocks.

- **Bug Fixes**
  - Fixed middleware configuration issues by implementing a more robust parameter passing system.

- **Documentation**
  - Updated function documentation to reflect new parameter structures and usage.

- **Style**
  - Removed unused imports and cleaned up code for better adherence to coding standards.

- **Versioning**
  - Incremented library version to reflect new changes and improvements.

### CODE REVIEW

File: sdks/aperture-go/sdk/middleware/http.go [hunk 1]
```diff
@@ -15,45 +13,34 @@ type HTTPMiddleware interface {
...
 		}
 
-		req := prepareCheckHTTPRequestForHTTP(r, m.client.GetLogger(), m.controlPoint, m.labels, m.rampMode)
+		req := prepareCheckHTTPRequestForHTTP(r, m.client.GetLogger(), m.middlewareParams.FlowParams)
 
-		flow := m.client.StartHTTPFlow(r.Context(), req, m.rampMode, m.timeout)
+		flow := m.client.StartHTTPFlow(r.Context(), req, m.middlewareParams)
 		if flow.Error() != nil {
 			m.client.GetLogger().Info("Aperture flow control got error. Returned flow defaults to Allowed.", "flow.Error()", flow.Error().Error(), "flow.ShouldRun()", flow.ShouldRun())
 		}
```
[the remark this answers is not in the document]
Comment (harjotgill) [reply sdks/aperture-go/sdk/middleware/http.go:46]: show me how to do it?

**Comment by hasit** [comment]
We should have `controlPoint` outside of `FlowParams`. How I think about it is that "start a flow with a name `controlPoint` and here are the `params` of it.

**Comment by harjotgill** [comment]
> We should have `controlPoint` outside of `FlowParams`. How I think about it is that "start a flow with a name `controlPoint` and here are the `params` of it.

Ok, we can probably change it for both ts and

**Comment by hasit** [comment]
We can merge it. I will make my changes on top it; that'll be easier.

File: sdks/aperture-go/sdk/middleware/http.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (hasit) [reply sdks/aperture-go/sdk/middleware/http.go]: [222c32c](https://github.com/fluxninja/aperture/pull/2939/commits/222c32cbc6718427f4d65d0f3160829c333d36a4) addresses this.

File: sdks/aperture-go/sdk/middleware/http.go [hunk 2]
```diff
@@ -15,45 +14,52 @@ type HTTPMiddleware interface {
...
+			if err != nil {
+				// Handle or log the error according to your error handling policy
+				// For example, you could log the error and continue without the problematic pattern
+				// Or you could return an error and fail the creation of the middleware
+			} else {
+				compiledIgnoredPaths[i] = compiledPattern
+			}
+		}
+		middlewareParams.IgnoredPathsCompiled = compiledIgnoredPaths
 	}
```
[the remark this answers is not in the document]
Comment (hasit) [reply sdks/aperture-go/sdk/middleware/http.go]: [22085c5](https://github.com/fluxninja/aperture/pull/2939/commits/22085c5419d428914ede73c4a3b30a172a19292e) addresses this.

**Review by tanveergill — APPROVED** [review]

**Pull request merged**